src/NuGetForUnity/Editor/PackageSource/NugetPackageSourceV3.cs

```diff
@@ -242,17 +243,22 @@ public List<INugetPackage> GetUpdates(
                                     searchQueryBuilder.Append($"packageid:{packagesToFetch[i].Id}");
                                 }
 
-                                updates.AddRange(
+                                async Task<List<INugetPackage>> SearchPackageAsync() =>
                                     await ApiClient.SearchPackageAsync(
                                             this,
                                             searchQueryBuilder.ToString(),
                                             0,
                                             0,
                                             includePrerelease,
-                                            CancellationToken.None)
-                                        .ConfigureAwait(false));
+                                            token)
+                                        .ConfigureAwait(false);
+
+                                fetchTasks.Add(SearchPackageAsync());
```

Comment:
No need to create a anonym function
```suggestion
                                fetchTasks.Add(ApiClient.SearchPackageAsync(
                                            this,
                                            searchQueryBuilder.ToString(),
                                            0,
                                            0,
                                            includePrerelease,
                                            token));
```

Reply:
Don't we need ConfigureAwait on each task? I am never sure about that... I'll test it like this on Monday and accept the change if it works out.

Reply:
When you don't have a `await` you don't wait for the task to be completed so you don't need to define on witch thread the code after the await (run when the task finished) should run.